### PR TITLE
MySQL statement binding

### PR DESCRIPTION
**Pull request by Joannis**
- [x] Adds statement binding for MySQL
- [x] Support more binding types
- [x] Support more result parsing types
- [x] Tests, comments and cleanup

### CODE REVIEW

File: Sources/MySQL/Query/Binding.swift [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    }
+    
+    /// Binds to an `UInt32` or `UInt64` depending on the processor architecture
+    ///
+    /// Binds to the first unbound parameter
+    ///
+    /// - throws: If the next unbound parameter is of a different type or if there are no more unbound parameters
+    public func bind(_ int: UInt) throws {
+        #if arch(x86_64) || arch(arm64)
+            try self.bind(numericCast(int) as UInt32)
```
Comment (tanner0101) [inline Sources/MySQL/Query/Binding.swift]: shouldn't this be flipped?

Comment (Joannis) [reply Sources/MySQL/Query/Binding.swift]: Good spot!

**Comment by Joannis** [comment]
@tanner0101 yes, the connection pool is redundant. I'm of the opinion that drivers (such as MySQL) should also provide a good experience. I don't think people are required to use an ORM like Fluent to get connection pooling. We could/should make a generic ConnectionPool. MongoDB, MySQL and other DBs are almost always the same. Some extra configurability here and there aside, the only real difference between protocols is that some DBs like MongoDB support multiplexing, and others don't.

File: Sources/MySQL/Query/Connection+Future.swift.orig [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import Async
+import Core
+
+extension Connection {
+    /// Collects all resulting rows and returs them in the future
+    ///
+    /// - parameter query: The query to be executed to receive results from
+    /// - returns: A future containing all results
+    internal func allRows(in query: Query) -> Future<[Row]> {
+<<<<<<< HEAD
```
Comment (tanner0101) [inline Sources/MySQL/Query/Connection+Future.swift.orig]: conflict

File: Sources/MySQL/Query/Connection+Query.swift.orig [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        
+        // SQL Query
+        buffer.append(0x16)
+        buffer.append(contentsOf: [UInt8](query.utf8))
+        
+        try self.write(packetFor: buffer)
+    }
+}
+
+<<<<<<< HEAD
```
Comment (tanner0101) [inline Sources/MySQL/Query/Connection+Query.swift.orig]: conflict

File: Sources/MySQL/Query/Query.swift.orig [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<<<<<<< HEAD
+/// Any query type
+=======
+/// http://localhost:8000/mysql/basics/#queries
+>>>>>>> 94165bfe45ef990a066b06af8dff2995ab3bf8b5
```
Comment (tanner0101) [inline Sources/MySQL/Query/Query.swift.orig]: tag

**Pull request merged**